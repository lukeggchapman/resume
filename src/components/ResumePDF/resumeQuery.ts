import { useStaticQuery, graphql } from 'gatsby'
import { Root } from 'hast-format'

export interface EducationData {
  edges: {
    node: {
      htmlAst: Root
      frontmatter: {
        degree: string
        school: string
        startDate: string
        endDate: string
        logo: {
          publicURL: string
        }
      }
    }
  }[]
}

export interface ExperienceData {
  edges: {
    node: {
      htmlAst: Root
      frontmatter: {
        role: string
        company: string
        startDate: string
        endDate: string
        logo: {
          publicURL: string
        }
      }
    }
  }[]
}

export interface SkillsData {
  edges: {
    node: {
      advanced: string[]
      intermediate: string[]
      beginner: string[]
    }
  }[]
}

export interface ResumeData {
  education: EducationData
  experience: ExperienceData
  skills: SkillsData
}

const resumeQuery = graphql`
  query {
    education: allMarkdownRemark(
      filter: { fields: { sourceInstanceName: { eq: "education" } } }
      sort: { fields: [frontmatter___endDate], order: DESC }
    ) {
      edges {
        node {
          htmlAst
          frontmatter {
            degree
            school
            startDate
            endDate
            logo {
              publicURL
            }
          }
        }
      }
    }
    experience: allMarkdownRemark(
      filter: { fields: { sourceInstanceName: { eq: "experience" } } }
      sort: { fields: [frontmatter___endDate], order: DESC }
    ) {
      edges {
        node {
          htmlAst
          frontmatter {
            role
            company
            startDate
            endDate
            logo {
              publicURL
            }
          }
        }
      }
    }
    skills: allSkillsJson {
      edges {
        node {
          advanced
          intermediate
          beginner
        }
      }
    }
  }
`

export const useResumeData = (): ResumeData => {
  return useStaticQuery(resumeQuery)
}

export default resumeQuery
