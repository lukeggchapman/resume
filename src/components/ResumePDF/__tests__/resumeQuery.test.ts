import { useStaticQuery } from 'gatsby'

import resumeQuery, { useResumeData } from '../resumeQuery'

const useStaticQueryMock = useStaticQuery as jest.Mock

describe('resume query', () => {
  it('should match snapshot', () => {
    expect(resumeQuery).toMatchSnapshot()
  })

  describe('useResumeData', () => {
    it('should return resume data', () => {
      const data = {
        some: 'data',
      }

      useStaticQueryMock.mockReturnValueOnce(data)

      expect(useResumeData()).toEqual(data)
      expect(useStaticQueryMock).toHaveBeenCalledTimes(1)
      expect(useStaticQueryMock).toHaveBeenCalledWith(resumeQuery)
    })
  })
})
