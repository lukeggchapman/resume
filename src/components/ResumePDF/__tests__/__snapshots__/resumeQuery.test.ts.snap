// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resume query should match snapshot 1`] = `
Array [
  "
  query {
    education: allMarkdownRemark(
      filter: { fields: { sourceInstanceName: { eq: \\"education\\" } } }
      sort: { fields: [frontmatter___endDate], order: DESC }
    ) {
      edges {
        node {
          htmlAst
          frontmatter {
            degree
            school
            startDate
            endDate
            logo {
              publicURL
            }
          }
        }
      }
    }
    experience: allMarkdownRemark(
      filter: { fields: { sourceInstanceName: { eq: \\"experience\\" } } }
      sort: { fields: [frontmatter___endDate], order: DESC }
    ) {
      edges {
        node {
          htmlAst
          frontmatter {
            role
            company
            startDate
            endDate
            logo {
              publicURL
            }
          }
        }
      }
    }
    skills: allSkillsJson {
      edges {
        node {
          advanced
          intermediate
          beginner
        }
      }
    }
  }
",
]
`;
